saleor/product/search.py

```diff
@@ -214,9 +214,27 @@ def get_search_vectors_for_values(
             )
             for value in values
         ]
+    elif input_type == AttributeInputType.REFERENCE:
+        # for now only AttributeEntityType.PAGE is supported
+        search_vectors += [
+            NoValidationSearchVector(
+                Value(get_reference_attribute_search_value(value)),
```

Comment:
This means: "one additional DB for each page", 
The task `update_products_search_vector_task` which calls implicitly this function has batch-size 300.  If you assume that each product has 4 references, we will end up with additional 1.2k queries.
I tested it with the product and 4 references. For 3.21 I have 13 db queries. After your changes I have 17. 


Additionally, would be good to have something like `page_id_to_page_title_map` 🤔 . This would allow you to fetch the reference's titles only once, it should also reduce the memory usage, as there is no need to fetch whole page object.